lib/src/androidTest/java/com/couchbase/lite/internal/core/C4QueryBaseTest.java

```diff
@@ -116,20 +118,25 @@ protected C4Query compile(String whereExpr, String sortExpr, boolean addOffsetLi
     protected List<List<List<Long>>> runFTS(Map<String, Object> params) throws LiteCoreException {
         List<List<List<Long>>> matches = new ArrayList<>();
         C4QueryOptions opts = new C4QueryOptions();
-        C4QueryEnumerator e = query.run(opts, encodeParameters(params));
+        AllocSlice encodedParams = encodeParameters(params);
+        C4QueryEnumerator e = query.run(opts,encodedParams);
         assertNotNull(e);
         while (e.next()) {
             List<List<Long>> match = new ArrayList<>();
             for (int i = 0; i < e.getFullTextMatchCount(); i++) { match.add(e.getFullTextMatchs(i).toList()); }
             matches.add(match);
         }
-        e.free();
+        if (e != null) e.free();
+        if (encodedParams != null) encodedParams.free();
         return matches;
     }
 
     private AllocSlice encodeParameters(Map<String, Object> params) throws LiteCoreException {
-        Encoder encoder = new Encoder();
-        encoder.write(params);
-        return encoder.finish();
+        FLEncoder encoder = new FLEncoder();
+        try {
+            encoder.write(params);
+            return encoder.finish2();
```

Comment:
is it going to be repetitive, if we add a comment saying to free return type? I feels like, any methods, which needs particular care with the return type should have a note saying so. So that if anybody looks at the method signature can see it, and can avoid the mistake. but repetition.

Reply:
I added the comment to AllocSlice and FLSliceResult itself.